activerecord/lib/active_record/encryption/configurable.rb

```diff
@@ -22,8 +22,6 @@ def configure(primary_key:, deterministic_key:, key_derivation_salt:, **properti
           config.deterministic_key = deterministic_key
           config.key_derivation_salt = key_derivation_salt
 
-          context.key_provider = ActiveRecord::Encryption::DerivedSecretKeyProvider.new(primary_key)
```

Comment:
I think you can move this line below the `properties.each...` loop and then do:

```
context.key_provider ||= ActiveRecord::Encryption::DerivedSecretKeyProvider.new(primary_key)
```

That way it will only build the key provider if not provided via config option.